oak/proto/permissions.proto

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak.permissions;
+
+// A PermissionsConfiguration provides a declarative description of the features that are permitted
+// for an Oak application.
+//
+// The permissions are applied dynamically at runtime, when creating the Nodes.
+message PermissionsConfiguration {
+  // Whether to enable the use of a GrpcServerNode as the front-door to the application.
+  bool grpc_front_door = 1;
+
+  // Whether to enable the use of an HttpServerNode as the front-door to the application.
+  bool http_front_door = 2;
+
+  // Whether to allow application logging. If true, allows creating a LogNode for the application.
+  bool logging = 3;
+
+  // Whether to allow monitoring (for the Runtime and the application). If true, the monitoring
+  // auxiliary service will be started.
+  bool monitoring = 4;
+
+  // Whether to allow introspection. If true, the introspection auxiliary service will be started.
```

Comment:
I guess this is only for the "debug" builds, not sure whether we need to specify it here though

Reply:
I thought it shoud be OK to allow introstpection for non-debug builds now that the node graph structure can no longer depend on non-public information.

Reply:
I am not sure either! :)

How about monitoring? Is that also something that can be enabled for non-debug builds? My understanding was that the same concerns apply to monitoring too. 

Does it make sense to allowlist the egress traffic here too (domains/IP to which we send the monitoring or introspection data)?

Reply:
I removed the flag for monitoring, but the flag for introspection is kept.